Sources/Rendering/Misc/TextureLODsDownloader/index.js

```diff
@@ -0,0 +1,185 @@
+import macro from 'vtk.js/Sources/macro';
+
+const { vtkErrorMacro } = macro;
+
+// ----------------------------------------------------------------------------
+
+const getRemoteFileSize = (url) => {
+  // This function only works if the server provides a 'Content-Length'.
+  // For some reason, the 'Content-Length' header does not appear to be
+  // given for CORS HEAD requests on firefox. So this will not work on
+  // firefox if the images do not have the same origin as the html file.
+  // TODO: figure out how to make this work for CORS requests on firefox.
+  return new Promise((resolve, reject) => {
+    let xhr = new XMLHttpRequest();
+    xhr.open('HEAD', url, true);
+
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState === 4) {
+        if (xhr.status === 200) {
+          const size = xhr.getResponseHeader('Content-Length');
+          resolve(size);
+        } else {
+          console.log('Failed to get remote file size of', url);
+          reject(xhr.status, xhr.statusText);
```

Comment:
resolve and reject expect only 1 argument. `reject(xhr)`.

Reply:
Fixed. Thanks Seb!